src/Build/Logging/BinaryLogger/Postprocessing/IBuildFileReader.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+
+namespace Microsoft.Build.Logging;
+
+public interface IBuildFileReader
+{
+    /// <summary>
+    /// An event that allows the caller to be notified when an embedded file is encountered in the binary log.
+    /// When subscriber is OK with greedy reading entire content of the file and is interested only in the individual strings (e.g. for sensitive data redaction purposes),
+    ///  it can simplify subscribing to this event, by using handler with same signature as handler for <see cref="IBuildEventStringsReader.StringReadDone"/> and wrapping it via
+    /// <see cref="ArchiveFileEventArgsExtensions.ToArchiveFileHandler"/> extension.
+    ///
+    /// <example>
```

Comment:
Do we usually put examples in the summary? I would have expected it in remarks.

Reply:
Oh - should not have been in summary. 

I'd probably still keep it in example instead of remarks. Inspiration from runtime: https://github.com/dotnet/runtime/blob/3e6612485e148eec314b10981b1e47324c9b5e27/src/libraries/Microsoft.Extensions.Configuration.CommandLine/src/CommandLineConfigurationExtensions.cs#L15-L65
What's your thought on that?

Reply:
As long as there's precedent I'm happy (until we go to publish docs and something blows up :-P)